src/WebJobs.Script/Rpc/JobHostLanguageWorkerChannelManager.cs

```diff
@@ -34,14 +38,20 @@ public void DisposeAndRemoveChannel(ILanguageWorkerChannel channel)
 
         public void DisposeAndRemoveChannels()
         {
+            Stopwatch timer = Stopwatch.StartNew();
             foreach (string channelId in _channels.Keys)
             {
                 if (_channels.TryRemove(channelId, out ILanguageWorkerChannel removedChannel))
                 {
+                    while (removedChannel.FunctionInputBuffers.Values.Sum(val => val.Count) > 0 && timer.ElapsedMilliseconds < _disposeGraceTimeInMilliseconds)
```

Comment:
were you able to test this E2E? We need to make sure having a while loop here does not block host/language worker from processing invocation responses

Reply:
Yup, I did. 

The way I did it was, I had a while(true) here and another infinite loop which kept loading the buffer with requests. Therefore upon triggering shutdown, dispose was stuck here in this loop and  LanguageWorkerChannel.InvokeResponse continued to get triggered due to the buffer being constantly loaded.

Hopefully that test covers it.